ui/src/app/views/cohort-builder/search/cohort-builder/cohort-builder.component.ts

```diff
@@ -164,55 +163,61 @@ export class CohortBuilderComponent implements OnInit, OnDestroy {
   }
 
   private updateCharts() {
-    let female = 0;
-    let male = 0;
-    let unknown = 0;
-    let aa = 0;
-    let ap = 0;
-    let c = 0;
-    let na = 0;
-    let h = 0;
-    let o = 0;
-    let u = 0;
+    /*
+     * TODO:
+     * Determine exactly how and what data is to be bundled with each Subject
+     * and whether decoding that data should take place in browser or in SQL.
+     * Until the exact encoding is finalized, this hack should generate some
+     * nice charts for a visual response to searches.
+     */
+    const genders = {
+      'Male': 0,
+      'Female': 0,
+      'Unknown': 0,
+    };
+    const races = {
+      'African American': 0,
+      'Asian/Pacific': 0,
+      'Caucasian': 0,
+      'Native American': 0,
+      'Hispanic': 0,
+      'Other': 0,
+      'Unknown': 0,
+    };
+
     this.totalSet.forEach((subject, index) => {
-      const values = subject.val.split(',');
-      if (values[1] === 'M') {
-        male++;
-      } else if (values[1] === 'F') {
-        female++;
-      } else {
-        unknown++;
+      const [uid, gender, race] = subject.val.split(',');
+      switch (gender) {
+          case '1': genders.Male++    ; break;
+          case '2': genders.Female++  ; break;
+          default : genders.Unknown++;
```

Comment:
Is the survey just asking for binary gender, or is there an `other` field? (This is also partially for my curiousity.)

Reply:
Same as race below, our gender options differ from this. Fine for a placeholder.